src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1207,23 +1197,36 @@ public void testDownloadCourseArchiveAsInstructor_not_found() throws Exception {
 
     // Test
     public void testDownloadCourseArchiveAsInstructor() throws Exception {
+        submissionRepo.deleteAll();
 
-        // Dummy course archive
-        Path courseArchivePath = Path.of(courseArchivesDirPath, "some-course-archive.zip");
-        if (!Files.exists(courseArchivePath)) {
-            Files.createDirectories(Path.of(courseArchivesDirPath));
-            Files.createFile(courseArchivePath);
-        }
+        // Archive the course and wait until it's complete
+        testArchiveCourseWithTestModelingAndFileUploadExercises();
 
-        // Generate a course that has an archive
-        Course course = database.createCourse();
-        course.setCourseArchivePath(courseArchivePath.toString());
-        course = courseRepo.save(course);
+        var course = courseRepo.findAll().stream().findFirst().get();
```

Comment:
Don't we have to assert that `findFirst` returned something before we `get`?

Reply:
We can but in this case I know that there's a course